ext/openssl/openssl.c

```diff
@@ -933,6 +933,46 @@ static void php_openssl_dispose_config(struct php_x509_request * req) /* {{{ */
 }
 /* }}} */
 
+static ENGINE *php_openssl_make_pkcs11_engine(const bool warn) /* {{{ */
+{
+	char *verbose = NULL;
+	ENGINE *engine;
+
+	engine = ENGINE_by_id("pkcs11");
```

Comment:
As already pointed in https://github.com/php/php-src/pull/6860#issuecomment-896684379 see https://www.openssl.org/docs/man3.0/man7/migration_guide.html#Providers-are-a-replacement-for-engines-and-low-level-method-overrides

Engines are replaced by providers, I'm not sure it makes sense to add compatibility with openssl1.1 ATM

Reply:
Please, can someone propose an update of this serie that use the providers instead of "engine" ? As stated, I am not working on this topic anymore. I was focused on it few months ago, but I cannot spend months in waiting.